logger/ringpop.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package logger
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/uber-common/bark"
+	"io/ioutil"
+)
+
+// Options is used to set the ringpop logger implementation and the minimum
+// output level for each named logger.
+type Options struct {
+	// The underlying logger implementation to use.
+	Logger bark.Logger
+
+	// Minimum level to which the Ringpop logger is restricted.
+	Ringpop Level
+	// Minimum level to which the Gossip logger is restricted.
+	Gossip Level
+	// Minimum level to which the Suspicion logger is restricted.
+	Suspicion Level
+	// Minimum level to which the Ring logger is restricted.
+	Ring Level
+	// Minimum level to which the Membership logger is restricted.
+	Membership Level
+	// Minimum level to which the Damping logger is restricted.
+	Damping Level
+}
+
+// LogFactory provides a convenient way to produce named loggers and pass
+// metadata as fields between separate parts of the codebase.
+type LogFactory interface {
+	// Ringpop retutrns the logger used to log messages outside of any module.
```

Comment:
You're right, there's no need for both. `Factory` is more convenient because you can create named loggers on the fly, `LogFactory` on the other hand has the benefit of being static. I think I was unsure on the rate we plan to use new named loggers and I didn't knew which one to keep.